File: spring-data-requery-test-java/src/main/java/org/springframework/data/requery/domain/sample/AbstractMappedType.java
package org.springframework.data.requery.domain.sample;

import io.requery.Generated;
import io.requery.Key;
import io.requery.Superclass;
import io.requery.Version;
import org.springframework.data.requery.domain.AbstractPersistable;

/**
 * AbstractMappedType
 *
 * @author debop@coupang.com
 * @since 18. 6. 25
 */
@Superclass
public abstract class AbstractMappedType extends AbstractPersistable<Long> {

    public AbstractMappedType() {}

    public AbstractMappedType(String attribute1) {
        this.attribute1 = attribute1;
    }

    @Key
    @Generated
    protected Long id;

    @Version
    protected Long version;

    String attribute1;


    private static final long serialVersionUID = 1828035517487100998L;
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/sample/AbstractMappedType.kt
package com.coupang.springframework.data.requery.domain.sample

import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.Generated
import io.requery.Key
import io.requery.Superclass
import io.requery.Version

/**
 * com.coupang.springframework.data.requery.domain.sample.AbstractMappedType
 * @author debop
 * @since 18. 5. 30
 */
@Superclass
abstract class AbstractMappedType: AbstractPersistable<Long>() {

    @get:Key
    @get:Generated
    abstract override val id: Long?

    @Version
    protected var version: Int = 0

    abstract var attribute1: String?

}